Fix cuboids / 3d / M6 (#320)
openvinotoolkit/datumaro: docs/formats/sly_pointcloud_user_manual.md

- [Format specification](#format-specification)
- [Import Supervisely Point Cloud](#import-supervisely-point-cloud-dataset)
- [Export to other formats](#export-to-other-formats)
- [Export to Supervisely Point Cloud](#export-to-supervisely-point-cloud)
- [Examples](#examples)

## Format specification

Point Cloud data format:
- [specification](https://docs.supervise.ly/data-organization/00_ann_format_navi).
- [example](https://drive.google.com/file/d/1BtZyffWtWNR-mk_PHNPMnGgSlAkkQpBl/view).

This dataset format supports the following types of annotations:
- `cuboid_3d`

Supported annotation attributes:
- `object` (read/write, integer),
- `createdAt` (write, string),
- `updatedAt` (write, string),
- `labelerLogin` (write, string), responsible for the corresponding fields
  in the annotation file.
- arbitrary attributes

Supported image attributes:
- `description` (read/write, string),
- `createdAt` (write, string),
- `updatedAt` (write, string),
- `labelerLogin` (write, string), responsible for the corresponding fields
  in the annotation file.
- `frame` (read/write, integer). Indicates frame number of the image.
- arbitrary attributes

## Import Supervisely Point Cloud dataset

> An example datast in Supervisely Point Cloud format is available for download:

Reviewer: ```suggestion
An example dataset in Supervisely Point Cloud format is available for download:
```

> <https://drive.google.com/u/0/uc?id=1BtZyffWtWNR-mk_PHNPMnGgSlAkkQpBl&export=download>

Reviewer: Where does this example come from?

Author: Actually, I don't know. @nmanovic, how did it get in SOW?

<!--lint disable fenced-code-flag-->
```
└─ Dataset/
    ├── ds0/
    │   ├── ann/
    │   │   ├── <pcdname1.pcd.json>
    │   │   ├── <pcdname2.pcd.json>
    │   │   └── ...
    │   ├── pointcloud/
    │   │   ├── <pcdname1.pcd>
    │   │   ├── <pcdname1.pcd>
    │   │   └── ...
    │   ├── related_images/
    │   │   ├── <pcdname1_pcd>/
    │   │   |  ├── <image_name.ext.json>
    │   │   |  ├── <image_name.ext.json>
    │   │   └── ...
    ├── key_id_map.json
    └── meta.json
```

There are two ways to import Supervisely Point Cloud dataset:

```bash
datum import --format sly_pointcloud --input-path <path/to/dataset>
# or
datum create
datum add path -f sly_pointcloud <path/to/dataset>
```

To make sure that the selected dataset has been added to the project,
you can run `datum info`, which will display the project and dataset
information.

## Export to other formats

Datumaro can convert Supervisely Point Cloud dataset into any other
format [Datumaro supports](../user_manual.md#supported-formats).

Such conversion will only be successful if the output
format can represent the type of dataset you want to convert,

> e.g. 3d point clouds annotations can be

Reviewer: ```suggestion
e.g. 3D point cloud annotations can be
```

> saved in `KITTI Raw` format, but not as `COCO keypoints`.

Reviewer: The inline code style seems inappropriate here. These aren't identifiers or code fragments.

> There are few ways to convert Supervisely Point Cloud dataset

Reviewer: ```suggestion
There are a few ways to convert Supervisely Point Cloud dataset
```

Author: "a few" would mean more than just "few". And we have exactly 2.

    -f kitti_raw -o my_kitti/ -- --save-images --reindex --allow-attrs
```

### Example 3. Create a custom dataset

``` python
from datumaro.components.dataset import Dataset
from datumaro.components.extractor import Cuboid3d, DatasetItem

dataset = Dataset.from_iterable([
    DatasetItem(id='frame_1',
        annotations=[
            Cuboid3d(id=206, label=0,
                position=[320.86, 979.18, 1.04],
                attributes={'occluded': False, 'object': 1, 'x': 1}),

            Cuboid3d(id=207, label=1,
                position=[318.19, 974.65, 1.29],
                attributes={'occluded': True, 'object': 2}),
        ],
        pcd='path/to/pcd1.pcd',
        attributes={'frame': 0, 'description': 'zzz'}
    ),

    DatasetItem(id='frm2',
        annotations=[
            Cuboid3d(id=208, label=1,
                position=[23.04, 8.75, -0.78],
                attributes={'occluded': False, 'object': 2})
        ],
        pcd='path/to/pcd2.pcd', related_images=['image2.png'],
        attributes={'frame': 1}
    ),
], categories=['cat', 'dog'])

dataset.export('my_dataset/', format='sly_pointcloud', save_images=True,
    allow_undeclared_attrs=True)
```

> More examples of working with KITTI Raw dataset from code can be found in

Reviewer: ```suggestion
More examples of working with Supervisely Point Cloud dataset from code can be found in
```
